### PR TITLE
[FIX] rating: correctly compute rating_last_value in RatingParentMixin

### PR DESCRIPTION
**Pull request by IT-Ideas**
Before this commit:
* rating_last_value takes non consumed ratings into account.

After this commit:
* rating_last_value only takes consumed ratings into account.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/project/tests/test_project_flow.py [hunk 1]
```diff
@@ -216,26 +216,26 @@ def test_rating(self):
 
         self.assertEqual(first_task.rating_count, 0, "Task should have no rating associated with it")
 
-        rating_good = self.env['rating.rating'].create({
+        rating_bad = self.env['rating.rating'].create({
```
Comment (tde-banana-odoo) [inline addons/project/tests/test_project_flow.py:219]: Is this change necessary ^^ ? Seems you are switching declaration but with same output ?

File: addons/rating/models/rating_mixin.py [hunk 2]
```diff
@@ -40,7 +40,9 @@ class RatingParentMixin(models.AbstractModel):
         compute='_compute_rating_percentage_satisfaction', compute_sudo=True, search='_search_rating_avg')
     rating_avg_percentage = fields.Float('Average Rating (%)', groups='base.group_user',
         compute='_compute_rating_percentage_satisfaction', compute_sudo=True)
-    rating_last_value = fields.Float('Rating Last Value', groups='base.group_user', related='rating_ids.rating')
+    rating_last_value = fields.Float(
+        'Rating Last Value',
+        groups='base.group_user', related='rating_ids.rating', domain="[('consumed', '=', True)]")
```
Comment (tde-banana-odoo) [inline addons/rating/models/rating_mixin.py:45]: Did not know domain was working on a related on a 2many field ^^ cool :D

File: addons/rating/models/rating_mixin.py [hunk 3]
```diff
@@ -40,7 +40,9 @@ class RatingParentMixin(models.AbstractModel):
         compute='_compute_rating_percentage_satisfaction', compute_sudo=True, search='_search_rating_avg')
     rating_avg_percentage = fields.Float('Average Rating (%)', groups='base.group_user',
         compute='_compute_rating_percentage_satisfaction', compute_sudo=True)
-    rating_last_value = fields.Float('Rating Last Value', groups='base.group_user', related='rating_ids.rating')
+    rating_last_value = fields.Float(
+        'Rating Last Value',
```
Comment (tde-banana-odoo) [inline addons/rating/models/rating_mixin.py:44]: Linting: we could make this a bit nicer to read, like
```
rating_last_value = fields.Float(
    'Rating Last Value', groups='base.group_user',
    related='rating_ids.rating', domain="[('consumed', '=', True)]")
```

**Review by tde-banana-odoo — COMMENTED** [review]
LGTM, some linting / side remarks :)

File: addons/project/tests/test_project_flow.py (hunk 1 above)
Comment (IT-Ideas) [reply addons/project/tests/test_project_flow.py:219]: Yep it is :) So that I do not have to change the whole test and just inverted the order of the non consumed and consumed.

File: addons/rating/models/rating_mixin.py (hunk 3 above)
Comment (IT-Ideas) [reply addons/rating/models/rating_mixin.py:44]: Okido 👍🏼

**Comment by IT-Ideas** [comment]
Thanks for the review @tde-banana-odoo 😘

**Review by xavierbol — APPROVED** [review]
LGTM, thanks for the fix 🙂

File: addons/rating/models/rating_mixin.py (hunk 2 above)
Comment (IT-Ideas) [reply addons/rating/models/rating_mixin.py]: Boarf, it actually works, but validation of field attributes is screaming on the runbot 😅

**Comment by IT-Ideas** [comment]
Closed as:
* not supported by the ORM and because:
    * computing it on the fly would be too costly
    * the search method would require using non ORM methods and using SQL
* the field is only used in project and PR #84781 removes that usage

=> #84784 has been created in master in order to remove the field.

**Comment by IT-Ideas** [comment]
Closed as:
* not supported by the ORM and because:
    * computing it on the fly would be too costly
    * the search method would require using non ORM methods and using SQL
* the field is only used in project and PR #84781 removes that usage

=> #84784 has been created in master in order to remove the field.
